src/metabase/models/setting.clj

```diff
@@ -652,6 +661,8 @@
   [setting-definition-or-name new-value]
   (let [{:keys [setter cache?], :as setting} (resolve-setting setting-definition-or-name)
         name                                 (setting-name setting)]
+    (when-not (current-user-can-access-setting? setting)
+      (api/throw-403))
```

Comment:
maybe ignore my previous comment. I kinda hate that settings will throw a 403 from a context that really isn't related to http at all. But maybe that's just the nature of it. But this feels off to me. No idea how much work it would be to clean it up but errors make sense, but throwing a 403 in setting code feels off to me.